libs/utils/src/decorators/is-msa-id.decorator.ts

```diff
@@ -1,5 +1,12 @@
 import { registerDecorator, ValidationArguments, ValidationOptions } from 'class-validator';
 
+export function isValidMsa(value: unknown): boolean {
+  // no leading zeros are accepted since this would cause different DsnpUserUri to be valid for the same user
+  const re = /^[1-9][0-9]*$/;
+
+  return (typeof value === 'string' && re.test(value)) || typeof value === 'number';
```

Comment:
Should probably validate that it doesn't exceed u64... either:
1. Convert to `bigint` and compare to 2^64
2. Add a max length to the RE for the longest-length allowable bigint string (less precise, but should be fine as a sanity check, as we're unlikely to ever get to the limit for a real MSA ID)